jck/update_git_material.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+# Now you can use these variables in your script
+JCK_ROOT_USED="$1"
+JCK_GIT_REPO_USED="$2"
+jck_branch="$3"
+isZOS="$4"
+echo "JCK_ROOT_USED is: $JCK_ROOT_USED"
+echo "JCK_GIT_REPO_USED is: $JCK_GIT_REPO_USED"
+echo "jck_branch is: $jck_branch"
+echo "isZOS is: $isZOS"
+
+delete_native_dirs() {
+    natives_directory="${JCK_ROOT_USED}/natives"
+    # Check if the directory exists
+    if [ -d "$natives_directory" ]; then
+        # Print a message indicating deletion
+        echo "Deleting $natives_directory..."
+        # Delete the directory and its contents
+        rm -r "$natives_directory"
+        # Check if deletion was successful
+        if [ $? -eq 0 ]; then
+            echo "Directory deleted successfully."
+        else
+            echo "Failed to delete directory: $natives_directory"
+        fi
+    else
+        echo "Directory does not exist: $natives_directory"
+    fi
+}
+
+# Function to perform git clone
+git_clone() {
+    # Set the target directory for cloning
+    clone_directory="${JCK_ROOT_USED}"
+    mkdir -p "$(dirname "$clone_directory")"
+    # Check if the parent directory was created successfully
+    if [ $? -eq 0 ]; then
+        # Change into the parent directory
+        cd "$(dirname "$clone_directory")" || exit 1
+        # Clone the repository into the target directory
+        git clone --single-branch -b "${jck_branch}" "${JCK_GIT_REPO_USED}" "$(basename "$clone_directory")"
+        # Check if the clone was successful
+        if [ $? -eq 0 ]; then
+            echo "Repository cloned successfully into $clone_directory"
+        else
+            echo "Failed to clone repository"
+            exit 1
+        fi
+    else
+        echo "Failed to create parent directory: $(dirname "$clone_directory")"
+        exit 1
+    fi
+}
+
+# Function to remove test material and git clone
+remove_and_clone() {
+    echo "Removing test material and git clone again"
+    rm -rf "$JCK_ROOT_USED"
+    git_clone
+}
+
+# Function to perform hard reset and pull
+perform_hard_reset_and_pull() {
```

Comment:
This should handle both zos and non-zos. We do not need perform_hard_reset_zos to handle zos separately.

Reply:
@llxia So I need to move the logic of `perform_hard_reset_zos` into  `perform_hard_reset_and_pull` based on ZOS flag or is it general logic which is already there? As I had inherited the logic from build.xml here, so not sure.

Reply:
Based on the ZOS flag, ZOS needs to replace the attribute file.